geomesa-process/geomesa-process-vector/src/main/scala/org/locationtech/geomesa/process/GeoMesaProcess.scala

```diff
@@ -24,18 +20,7 @@ trait GeoMesaProcess extends VectorProcess
 /**
   * Common trait for visitors, allows for feature collections to execute processing in a standardized way
   */
-trait GeoMesaProcessVisitor extends FeatureCalc with FeatureAttributeVisitor {
-
-  /**
-    * Optimized method to run distributed processing. Should set the result, available from `getResult`
-    *
-    * @param source simple feature source
-    * @param query may contain additional filters to apply
-    */
-  def execute(source: SimpleFeatureSource, query: Query): Unit
-
-  // hook to allow delegation from retyping feature collections
-  override def getExpressions: java.util.List[Expression] = Collections.emptyList()
-}
+@deprecated("Replaced with org.locationtech.geomesa.index.process.GeoMesaProcessVisitor")
```

Comment:
Is this an API change which needs to be documented in the upgrade guide?

Reply:
Hmm, we haven't been documenting deprecated functions so far - instead we document when they actually get removed. Documenting the deprecated functions is probably a good idea, but maybe we can address it in a different PR since there are lots of other ones as well.

Reply:
https://geomesa.atlassian.net/browse/GEOMESA-2460